pkg/controller/predicates/resources/resources.go

```diff
@@ -53,3 +58,41 @@ func Deleted() predicate.Funcs {
 		},
 	}
 }
+
+// Content predicates moved from original controller.
+var CMContentChangedPredicate = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldCM, _ := e.ObjectOld.(*corev1.ConfigMap)
+		newCM, _ := e.ObjectNew.(*corev1.ConfigMap)
+		return !reflect.DeepEqual(oldCM.Data, newCM.Data)
+	},
+}
+
+var SecretContentChangedPredicate = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldSecret, _ := e.ObjectOld.(*corev1.Secret)
+		newSecret, _ := e.ObjectNew.(*corev1.Secret)
+		return !reflect.DeepEqual(oldSecret.Data, newSecret.Data)
+	},
+}
+
+var DSCDeletionPredicate = predicate.Funcs{
```

Comment:
this may not be doing what it is expected as by default all the methods return `true` : https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.17/pkg/predicate/predicate.go#L54-L99

Reply:
this newly added code seems a mistake.
it is not used anywhere,
a similar one was already in the DSCI,  
but you are right.
we probaby no need this predicate on DSC deletion at all. r.watchDSCResource is already dealt with that case.